[FEAT] KakaoLogin 구현_재PR

HyeonChae1104 opened the pull request:
## 주요 변경사항 ✔️
### 로그인 로직
1. 로그인은 get 요청을 통해 백엔드의 카카오로그인으로 redirect됩니다. 
2. 백엔드는 쿠키에 refresh token만 httponly로 넣어주고 로그인 성공 페이지( `SuccessPage`)로 redirect 시킵니다.
3. 프론트는 성공 페이지( `SuccessPage`)에서 accessToken 재발급 요청을 보내는 방식을 사용합니다.

- `Login/index.tsx` : 백엔드로 get 요청
-  `FailurePage`: 로그인 실패 시 리다이렉트 되는 페이지
 - `SuccessPage` : 로그인 성공 시 리다이렉트 되는 페이지

## 리뷰어에게... 🙏
401 응답에 대한 인터셉터 부분은 `instance/index.ts` 파일에 추가해도 되는지 여쭤보고 수정해보려고 남겨두었습니다.! -> 나머지 부분에 문제가 없다면, 마이페이지 구현 후 새로 이슈를 파서 수정해보겠습니다. 

그 이외의 리뷰해주신 부분은 반영을 해보았습니다. 잘못된 부분이 있는지 확인 부탁드립니다.! 

제 git에서는 충돌 파일이 없다고 뜨는데 확인 한번만 부탁드립니다.!ㅜ 있다면 다연님 말씀대로 빠르게 새 브랜치를 파서 코드 복붙 하도록 하겠습니다.

## 관련 이슈
closes #40

HyeonChae1104 committed:
feat(onboarding): 온보딩 화면에 들어갈 이미지와 아이콘 추가

- 온보딩 화면에서 서비스의 기능 설명을 위한 사진을 추가하였습니다.
- 카카오 로그인 버튼을 위한 아이콘을 추가하였습니다.

#31

HyeonChae1104 committed:
feat(onboarding): 온보딩 화면을 구현

- 온보딩 과정에 사용할 이미지를 추가
- 제목, 부제목, 카카오 로그인 버튼을 추가
- 기능 설명을 위해 화살표를 통한 네비게이션 기능을 구현했습니다.

#31

HyeonChae1104 committed:
feat(router): 온보딩 화면을 위한 라우트 설정 추가

- 온보딩 화면으로 이동할 수 있는 라우트를 추가했습니다.

#31

HyeonChae1104 committed:
chore(onboarding): 폰트 변경

폰트를 Pretendard로 변경하였습니다.

#31

HyeonChae1104 committed:
feat(onboarding): 작은 화면에서 적용되는 반응형 디자인 추가

- 768px 이하의 화면에서 이미지가 위로 올라가도록 @media 쿼리 추가
- 화면이 작을 때 레이아웃이 수직으로 정렬되도록 수정하였습니다.

#31

HyeonChae1104 committed:
feat: 카카오로그인버튼 공통 컴포넌트 추가

카카오로그인 버튼은 온보딩 화면과 로그인 로직에서 재사용하기 위해 공통 컴포넌트로 분리하였습니다.

#40

HyeonChae1104 committed:
feat:(Login): 백엔드로 리다이렉트하는 카카오 로그인 기능 추가

카카오 로그인 버튼을 통해 백엔드로 리다이렉트하는 기능을 추가하였습니다.(GET 요청)

#40

HyeonChae1104 committed:
feat(success-page): 액세스 토큰 유효성 검증 및 홈화면 리다이렉트 로직 추가

액세스 토큰을 확인하여 유효하면 홈으로 이동, 그렇지 않으면 실패 페이지로 리다이렉트합니다.

#40

HyeonChae1104 committed:
feat(failure-page): 리프레시 토큰을 통한 재로그인 기능 및 카카오 로그인 버튼 스타일 재사용 추가

에러 메시지를 표시하고, "다시 로그인하기" 버튼을 눌러 리프레시 토큰을 사용해 다시 액세스 토큰을 받아 성공 페이지로 이동합니다.

#40

HyeonChae1104 committed:
feat(button): children 속성 추가 및 카카오 로그인 버튼 컴포넌트 개선

- children 속성 추가로 버튼 안에 동적으로 텍스트 삽입 가능
- 버튼 디자인 재사용성 강화
- 스타일 컴포넌트 유지하며 클릭 이벤트 핸들링 기능 보완하였습니다.

#40

HyeonChae1104 committed:
feat(routes): 성공 및 실패 페이지 경로 추가

성공/실패 페이지 경로 추가 및 경로 설정을 하였습니다.

HyeonChae1104 committed:
feat(routes): 성공 및 실패 페이지 경로 추가

성공/실패 페이지 경로 추가 및 경로 설정을 하였습니다.

#40

HyeonChae1104 committed:
refactor(SuccessPage): SuccessPage의 로직을 훅으로 분리

- getCookieValue를 통해 accessToken을 쿠키에서 가져옵니다.
-accessToken 확인 후 localStorage에 저장하고, 보안을 위해 그 값을 바로 삭제합니다.
- accessToken이 존재하면 홈 페이지(/)로 이동하고, 존재하지 않으면 실패 페이지(/failure)로 이동합니다.

#40

HyeonChae1104 committed:
refactor(FailurePage): FailurePage의 로직을 훅으로 분리

- named export 방식으로 수정
- 백엔드에서 로그인 실패할 경우 redirect 되는 페이지로 '카카오로그인버튼'을 누르면 다시 백엔드의 로그인 로직으로 넘어갈 수 있도록 합니다.
- 에러 상황을 처리했던 로직은 프론트에서 불필요하여 전부 삭제하였습니다.

HyeonChae1104 committed:
refactor(FailurePage): FailurePage의 로직을 훅으로 분리

- named export 방식으로 수정
- 백엔드에서 로그인 실패할 경우 redirect 되는 페이지로 '카카오로그인버튼'을 누르면 다시 백엔드의 로그인 로직으로 넘어갈 수 있도록 합니다.
- 에러 상황을 처리했던 로직은 프론트에서 불필요하여 전부 삭제하였습니다.

#40

HyeonChae1104 committed:
feat(authService): 401 응답 시 토큰 재발급을 위한 useFetchWithToken 추가

- accessToken 만료 시 토큰 재발급을 처리하는 useFetchWithToken 훅 생성
- Axios 인터셉터를 통해 토큰 재발급 후 요청을 자동 재시도하도록 구현
- 토큰 재발급 실패 시 로그인 페이지로 리다이렉트

#40

HyeonChae1104 committed:
fix(login):

-  login 로직 변경에 의한 import 경로 추가
- named export로 통일

#40

HyeonChae1104 committed:
refactor(useSuccessLogin): 로그인 성공 시 쿠키 처리 로직 추가

- 로그인 성공 후 토큰 처리 기능 구체화하였습니다.

HyeonChae1104 committed:
Merge branch 'Weekly9' into feat/40-login-page

geongyu09 commented on src/api/hooks/useFetchWithToken.ts:
```diff
@@ -0,0 +1,37 @@
+}
+
+export const useFetchWithToken = () => {
+  const navigate = useNavigate();
```
유저레벨의 훅을 api 레벨에서 사용하는건 올바르지 않아보입니다. 
useNavigate는 컴포넌트에서 사용하는걸로 수정해야할 것 같아요

실제 api 에서 요청하는 로직만 있는게 아니라 리다이렉트까지 책임을 가지게 되면, 나중에 이해할 수 없는 동작을 하는데 원인을 못찾는 경우가 발생합니다. 
흔히 이야기 하는 사이드이팩트가 이런 경우입니다

추후 혼란 방지를 위해서 책임은 하나만 가지도록 하는게 좋아보입니다

geongyu09 replied:
추가적으로 인터셉터가 훅 내부에 있는건 이전까지 저희가 작성해왔던 계층구조와 맞지 않는 것 같습니다. 
이전의 코드에서는 인터셉터는 instance에 존재합니다. 
해당 로직도 instance에 있어야 할 것 같습니다. 

그리고 해당 파일에는 동일한 인스턴스의 인터셉터가 존재하므로 동작 또한 겹치게 될 것 같습니다. 
해당 인스턴스는 현재 어플리케이션 전체적으로 사용중이므로 꼭 문제가 없도록 만들어주세요

HyeonChae1104 replied:
401 응답에 처리하는 부분은 `instance/index.ts` 파일에 추가하는 방식으로 진행하면 될까요?

geongyu09 replied:
넵 적절히 추가해주시면 좋을 것 같아요

Da-Yeon-Kim commented:
저녁에 집 가서 살펴보겠습니다! 저도 로그인 로직을 잘 몰라서 열심히 봐봐야 할 것 같아요😅
슬쩍 봐보니 merge conflict 정리가 덜 된 것 같은데, 일단 리뷰 먼저 받으려고 미완인 채로 올려두신 건가요??

geongyu09 commented on src/api/hooks/useHandleKakaoLogin.ts:
```diff
@@ -0,0 +1,19 @@
+      navigate('/login');
+    } else {
+<<<<<<< HEAD
+      window.location.href ='url';  //추후 수정
```
여기는 왜 navigate를 안쓰고 window를 사용하셨나요? 
혹시 그 이유가 있을까요?

HyeonChae1104 replied:
window를 사용해서 get요청을한 이유는 백에서 get 요청에 대한 응답이 아닌 ,카카오로그인으로 redirect를 시키는 로직을 사용하셨습니다. 따라서 그 자리에서 무한히 기다리는 것이 아닌 응답을 기다리지 않는 `window.location.href`를 사용하였습니다.

HyeonChae1104 replied:
해당 로직은 `Login/index.tsx` 파일로 옮겼습니다!

geongyu09 commented on src/api/hooks/useHandleKakaoLogin.ts:
```diff
@@ -0,0 +1,19 @@
+      navigate('/login');
+    } else {
+<<<<<<< HEAD
+      window.location.href ='url';  //추후 수정
```
해당 로직은 api 의 성격과 맞지 않아보이네요.
이전에 저희가 짜왔던 로직에는 url을 가져오는 로직과 해당 url로 fetch를 날리는 axios문, 그리고 이를 스케줄링하는 query문이 있습니다.

geongyu09 commented on src/api/hooks/useSuccessLogin.ts:
해당 파일도 api 로직과는 성격이 맞지 않아보입니다. 
다른 곳으로 옮길 필요가 있어보이네요

geongyu09 replied:
프론트단에서 쿠키를 만지는 것은 문제가 있습니다. 
쿠키를 프론트에서 만질 수 있다는 건, 제 3자가 이를 이용할 수 있다는 말과도 동일합니다
쿠키의 모든 책임은 서버에 두는 것이 맞습니다. 

만약 단순히 토큰을 주고 받는 것이라면 이를 로컬스토리지에 저장하는것이 조금 더 바람직해보이네요.

백엔드에서 쿠키로 넣어주는 상황이라면 프론트에서 만질 수 없도록 설정해주라고 해야합니다

HyeonChae1104 replied:
백과 협의 후 방식을 바꾸었습니다! 감사합니다

geongyu09 commented on src/api/hooks/useHandleKakaoLogin.ts:
```diff
@@ -0,0 +1,19 @@
+import { useNavigate } from 'react-router-dom';
+
+export const useHandleKakaoLogin = () => {
```
혹시 핸들러를 다른 파일로 뺀 이유가 있으신가요?
저의 예상으로는, 다른 곳에서 해당 로직을 재사용 하는 경우가 없을 것으로 보입니다.
만약 재사용하기 위해서 훅으로 만들어야한다면 도메인을 제외하고 공통 로직부분만 훅으로 만들어야 하지 않을까 싶어요

HyeonChae1104 replied:
저도 하다 보니 다른 부분에서는 필요가 없을 것 같아 파일을 삭제하였습니다.!

geongyu09 commented on src/pages/Onboarding/index.tsx:
```diff
@@ -27,7 +28,11 @@ const onboardingDescriptions = [
 
-const OnboardingPage: React.FC = () => {
+interface OnboardingPageProps {
+  handleKakaoLogin: () => void; // props로 로그인 로직을 받음
```
혹시 페이지 컴포넌트에서 props를 받아야 하는 이유가 있나요?
아마 최상위 컴포넌트가 page컴포넌트일텐데, 그 위에서 라우팅을 만들어주는 로직에서 굳이 넣어줘야할 이유가 있을까 싶어요. 
그것도 버튼 컴포넌트에서만 사용되는 이밴트 핸들러인데, 그 위에 있는 페이지 컴포넌트도 이 로직을 알게 되고, 의미 없이 내려주기만 하고 있어요. 
혹시 이렇게 작성한 의도가 있을까요?

HyeonChae1104 replied:
해당 부분도 반영해두었습니다.! 꼼꼼하게 봐주셔서 감사합니다!

geongyu09 commented on e --abort:
해당 파일이 커밋에 들어간 이유가 있나요....?
해당 파일을 생성한 이유는 무엇인가요? 🤔

HyeonChae1104 replied:
삭제하였습니다!

geongyu09 reviewed (changes requested):
말씀 하신대로 전체적으로 충돌이 해결되지 않은 것 같아요.
충돌을 전부 다 해결하고, 돌아가는 코드가 되어야 올바르게 리뷰를 할 수 있을 것 같아요..

깃 관련된 내용을 제외하고 로직 부분으로만 리뷰를 남겨보았습니다. 
충돌 해결하고 몇가지 수정된 이후에 다시 재요청 해주세요!

HyeonChae1104 committed:
chore: 불필요한 파일 삭제

rebase 중 생긴 불필요한 파일을 삭제하였습니다.

#40

Da-Yeon-Kim commented:
git 해결 되셨을까요?!
혹시 git이 제대로 꼬여서 해결이 어렵다면, 그냥 코드는 따로 복붙 해두고
아예 Weekly9에서 새로 브랜치 만들어서 다시 PR 날리는 것도 나쁘지 않아 보여요.....
git은 git으로 해결하는 게 좋긴 하지만, 지금은 시간 상😥

HyeonChae1104 committed:
chore: 불필요한 파일 삭제

로그인 로직의 불필요한 파일을 삭제하였습니다.

#40

HyeonChae1104 committed:
Merge branch 'feat/40-login-page' of https://github.com/kakao-tech-campus-2nd-step3/Team16_FE into feat/40-login-page

HyeonChae1104 committed:
refactor(SuccessPage): 로그인 성공 시 redirect 되는 페이지 구현

- API 요청으로 액세스 토큰 재발급 및 로컬 저장소에 저장
- 토큰 재발급 후 홈 페이지로 자동 이동합니다.

#40

HyeonChae1104 committed:
refactor(FailurePage): 로그인 실패 시 redirect 되는 페이지 구현

- 로그인 실패 시 에러 문구
- 로그인 버튼을 누르면 다시 백엔드 로그인 로직으로 redirect 되도록 설정하였습니다.

#40

HyeonChae1104 committed:
refactor(LoginPage): 로그인 페이지 기능 수정

- 로그인 버튼을 통해 백엔드 로그인로직으로 redirect 합니다.

#40

HyeonChae1104 committed:
refactor(OnboardingPage): 온보딩 페이지에서 로그인 페이지로 navigate 기능 추가

#40

HyeonChae1104 committed:
fix: routes 파일에서 경로 수정

온보딩 경로 수정하여 로그인 페이지로 올바르게 이동하도록 변경

#40

HyeonChae1104 committed:
fix: 리뷰 반영 및 코드 수정

- 온보딩 페이지의 props를 제거하였습니다.
- 라우터 파일을 기존의 방식으로 변경
- 카카오로그인 버튼의 children 속성을 제거하였습니다.

#40

geongyu09 commented on src/pages/Login/FailurePage.tsx:
```diff
@@ -0,0 +1,17 @@
+export const FailurePage: React.FC = () => {
+  const login = () => {
+    const baseURL = process.env.REACT_APP_API_URL
+    window.location.href = `${baseURL}/login`
```
혹시 해당 로직이 어떤 기능을 하는지 설명해주실 수 있나요..?
문장만 보면 api 요청에 사용되는 기초 url에 /login으로 리다이렉트 시키는 로직같은데, 
제가 이해가 잘 안되는 부분이 있는 것 같아요

HyeonChae1104 replied:
백에서 `@GetMapping("/login")` 코드를 쓰셔서 저도 `window.location.href = `${baseURL}/login` `으로 구현을 하였습니다!

+ /login으로 엔드포인트를 사용하신걸로 판단했습니다!

HyeonChae1104 replied:
하지만 제 생각으로는 FailurePage는 거의 사용되지 않을 것으로 예상됩니다.

카카오로그인 서버나 잘못된 인증코드 등의 문제가 생길경우 자체적으로 처리를 한다고 알고 있고, 
토큰 만료나 유효하지 않은 경우 401 응답이 나오게 되고, 그 부분은 `instance/index` 파일에 추가할 것이기 때문에 이 페이지는 사용되지 않을 것 같습니다. 

하지만 백에서 요청해주신 부분이라 `Login/index.tsx` 에서 사용한 로직을 그대로 옮겨와서 구현했습니다!

geongyu09 replied:
아무튼 api 요청을 하는것이 아니라 리다이랙트를 하는것이 맞는 것이죠..?

추가적으로 baseURL은 이미 만들어져있습니다. 해당 값을 사용해도 좋아보이네요

geongyu09 commented on src/pages/Login/SuccessPage.tsx:
```diff
@@ -0,0 +1,40 @@
+      if (response.status === 200) {
+        console.log('토큰 재발급 성공:', response.headers);
+        const newAccessToken = response.headers.accessToken;
+        localStorage.setItem("accessToken", newAccessToken); 
```
토큰은 로컬에 저장하는 방식으로 결정하신건가요??
그렇다면 이전에 만들어두었던 storage를 사용하는게 맞을 것 같아요
아마 로컬스토리지코드는 작성 안되어있을텐데 작성 후에 사용하는게 좋아보이네요

Da-Yeon-Kim replied:
참고로 storage 코드는 `@/utils/storage`에 있습니다. 😊

HyeonChae1104 replied:
원래대로 토큰은 로컬에 저장하는 방식을 사용하기로 하였습니다! 기존의 `@/utils/storage` 파일에서 localStorage코드를 추가하도록 하겠습니다.!

감사합니다!

geongyu09 commented on src/pages/Login/SuccessPage.tsx:
```diff
@@ -0,0 +1,40 @@
+
+  useEffect(() => {
+    // 페이지에 들어오면 reissueAccessToken 함수 실행
+    reissueAccessToken().then(() => {
```
이 부분도 잘 이해를 못하였는데요.. 
성공 페이지에서 토큰 발급이 아니라 재발급을 하는거면, 이전에 토큰을 발급하는 절차가 또 있나요?

HyeonChae1104 replied:
백에서 로그인이 성공하면 쿠키에 refresh token만 담아 해당 페이지로 redirect 시킵니다. 
그럼 프론트는 `reissueAccessToken` 함수를 호출하여 쿠키를 포함한 요청을 다시 백으로 보내서 accessToken을 발급 받는 과정입니다!

따라서 재발급이라고 표현하였습니다!

geongyu09 replied:
설명 감사합니다! 이 상황이라면 재요청이라고 하는게 맞는 표현 같네요

geongyu09 commented on src/pages/Login/SuccessPage.tsx:
```diff
@@ -0,0 +1,40 @@
+  const navigate = useNavigate(); // useNavigate 훅을 사용하여 페이지를 이동합니다.
+
+  const reissueAccessToken = async () => {
+    try {
```
axios문이 컴포넌트에 있고, 이전에 만들어둔 인스턴스를 사용하지 않고있고, 요청 api 문을 useEffect를 사용하여 수행하는 과정으로 작성하셨는데,
기존에 작업을 해왔던 계층구조와 추상화 레벨을 어기게 된다면, 그에대한 합당한 이유가 있어야 한다고 생각해요

다른 팀원들을 설득할만한 부분이 혹시 있나요? 작성하신 이유가 궁금합니다!

Da-Yeon-Kim replied:
이게 무슨 말씀이시냐면, 저희는 지금 `@/api/hooks` 폴더에 백과 API 소통을 하는 코드들을 따로 분리해 관리하고 있어요.  `@/api/instance` 파일에 이미 만들어둔 axois문과 instance가 있고, 다른 훅들은 이를 활용한 `fetchWithToken`을 사용합니다. 그리고 스케줄링은 `useQuery`를 통해서 하고 있고요!

그런데 지금 이 SuccessPage에서는 기존에 만들어둔 걸 활용 안 하고, 직접 axois문부터 구현하고 계셔서 여쭤보는 겁니다! (그리고 분리도 되어있지 않고요!)

HyeonChae1104 replied:
오.. 맞네요.. 자세하게 설명도 해주셔서 감사합니당..ㅜ
두 분 말씀대로 코드를 수정하는게 좋을 것 같습니다.!

이 부분부터 수정해보겠습니다!

geongyu09 reviewed (commented):
고생이 많으시네요
코드 관련해서 커맨트 남겼습니다
시간이 얼마 남지 않은 시점에서 돌아가는 코드를 수정해달라는건 조금 부담이 될 수밖에 없으니, 
재요청 부분에 대한 답변을 듣고 approve하도록 하겠습니다!

Da-Yeon-Kim reviewed (commented):
저도 토큰 쪽이 이해가 잘 안 가는데 한 번만 동작이 흘러가는 과정을 설명해주실 수 있을까요? 그리고 `onboardingPage`와 `LoginPage`의 기능이 무엇이 다른지도 잘 모르겠어요😥 왜 `LoginPage`가 따로 있는 건가요?!

geongyu09 reviewed (approved):
제 의문점은 다 해결되어서 일단 approve하겠습니다! 
이후에 코드상의 문제는 연결하면서 수정하면 되지 않을까 싶습니다

Da-Yeon-Kim reviewed (approved):
저도 일단은 approve 해두겠습니다!
헷갈리는 저를 위해사 `onboardingPage` 외에 `LoginPage`가
따로 있는 이유만 살짝 설명 부탁 드려요😥

HyeonChae1104 commented:
어이고..! 제가 생각을 해보느라 답변이 늦어졌네요.! 죄송합니다
다연님이 로그인 페이지가 존재하는 이유에 대해 궁금해하신 점이 저도 이해가 됩니당,,

저도 질문을 받고 고민을 해봤는데, 로그인페이지가 사용되는 일은 2가지였습니다.! 온보딩 페이지에서 로그인 버튼을 통해 로그인 페이지로 넘어가는 것, 401 응답을 받아 SuceessPage에서 refresh-token을 재발급 받는 것을 시도하고 실패하면 로그인 페이지로 넘어가도록 생각을 했었습니다.

하지만 두 과정 모두 불필요한 작업이라고 생각이 드네요.!! 
온보딩 화면에서 로그인 버튼을 통해 바로 백엔드 로그인 로직으로 넘어가도록 수정하고, 401 응답을 받은 상황에서 토큰 재발급에 실패할 경우에도 백엔드 로그인 로직으로 바로 넘어가는 걸로 수정하는게 좋을 것 같습니다!

꼼꼼하게 살펴봐주셔서 감사합니다!!

HyeonChae1104 committed:
feat(SuccessPage): SuccessPage의 API 호출 로직 분리 및 fetchWithToken 설정 반영

- SuccessPage에서 API 호출 로직을 분리
- fetchWithToken 인스턴스를 사용하여 Authorization 헤더 및 baseURL 설정을 적용하였습니다.

#40

HyeonChae1104 committed:
feat(storage): 로컬 스토리지에 accessToken 저장 기능 추가

- util/storage에 authLocalStorage 추가
- reissueAccessToken 함수에서 authLocalStorage 사용하여 accessToken 저장

#40

HyeonChae1104 committed:
refactor: 온보딩 화면에서 로그인 페이지 거치지 않고 바로 백엔드 로그인으로 이동하도록 수정

- 온보딩 화면의 카카오 로그인 버튼 클릭 시 백엔드 로그인 엔드포인트로 바로 이동
- LoginPage 컴포넌트 및 관련 라우트 삭제하여 불필요한 중간 단계 제거하였습니다.

#40

HyeonChae1104 committed:
feat(instance): 401 Unauthorized 응답 시 토큰 재발급 및 로그인 리다이렉트 기능 추가

- fetchWithToken 인터셉터에 401 응답 처리 로직 추가
- 401 Unauthorized 응답 시 SuccessPage로 리다이렉트하여 토큰 재발급 시도
- 재발급 실패 시 백엔드 로그인 페이지로 리다이렉트하여 재로그인 요구하도록 구현하였습니다.

#40

HyeonChae1104 commented:
어제 리뷰 주신 내용을 전부 반영해보았습니다.
머지 전에 확인 부탁드립니다!

geongyu09 commented on src/pages/Login/SuccessPage.tsx:
```diff
@@ -0,0 +1,24 @@
+    if (status === 'success') {
+      navigate(RouterPath.home); // 성공 시 홈 페이지로 이동
+    } else if (status === 'error' && error) {
+      console.error('토큰을 가져오던 중 에러 발생', error);
```
이전보다 훨씬 좋은 코드가 되었네요

추가적으로 고민(만)해볼 부분 몇가지 남겨보면,
1. 성공 페이지에서 토큰 발급시에 에러가 발생하게 되면 콘솔만 찍고 끝나게 되어야 할까요? 아니면 유저 입장에서 토큰 발급에서 에러가 발생하면 어떤 화면을 보게 될까요? 
2. 재발급 요청은 언제 해야할까요? 단순히 로그인할때 토큰을 얻고자 하는것은 아닐거에요. 추후에 토큰이 만료 되었을 때 refrechToken이 있다면 다시 accessToken을 받을때 주로 사용할텐데, 이걸 어느계층에서 하는게 좋을까요?

geongyu09 reviewed (approved):
확인했습니다
이후의 수정은 백엔드와 헙업하면서 고쳐나가도 충분해 보입니다
